src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -623,7 +623,8 @@ bool primitive_inst::need_reset_output_memory() const {
             auto feature_block_size = get_feature_block_size(fmt);
             // if layout is single blocked and feature size is not aligned with the blocking size, need to reset output so that we can guarantee zero-filling
             // NOTE: We may improve this logic to avoid reset if we are sure that it is not "corrupted" by other layers.
-            if (output_layout.feature() % feature_block_size != 0) {
+            auto feature_dim = output_layout.get_partial_shape()[1];
+            if ((feature_dim.is_dynamic()) || (feature_dim.is_static() && feature_dim.get_length() % feature_block_size != 0)) {
```

Comment:
nit: As far as I know, if the first condition in || is positive, the next one won't be checked, so this can be simplified
```suggestion
            if (feature_dim.is_dynamic() || feature_dim.get_length() % feature_block_size != 0) {
```